bluetooth: controller: Include nrf_clock_control.h when MPSL is used

sofu1 opened the pull request:
Enable use of defines in nrf_clock_control.h when MPSL clock driver
is used in Zephyr bluetooth LL.

Signed-off-by: Solveig Fure <solveig.fure@nordicsemi.no>

Rallare committed:
[nrf noup]: Add LTE family and protocol.

Add AF_LTE family and NPROTO_AT protocol define for nRF91 in net_ip.h.

Signed-off-by: Håkon Alseth <haal@nordicsemi.no>
Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>
Signed-off-by: Marti Bolivar <marti.bolivar@nordicsemi.no>
(cherry picked from commit 821c2cbccf9b12bf1c55d117a1a7d9f188de46a8)
(cherry picked from commit f507dc67d480ceaf860f19678f42bce92f55b69b)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

Commit:
[nrf noup] ci: Add Jenkins-based CI for zephyr

Build on Jenkins

Signed-off-by: Ulrich Myhre <ulrich.solli.myhre@nordicsemi.no>
(cherry-picked from commit d9eb9fdf6018aba676fefc27e3047fec30dea6da)
(cherry picked from commit 66b52e33b66531b5ea8f48ffd9df05512d7126a1)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

sigvartmh committed:
[nrf temphack] include: arm: linker.ld

Fix up for linker script to place firmware metadata in the built hexfile
for secure bootloader.

Signed-off-by: Sigvart Hovland <sigvart.hovland@nordicsemi.no>
(cherry picked from commit 0be5514e484469992eb0dcec1ea58133b988ea5b)
(cherry picked from commit 3efa0803437d978fc259141fc03f732de09342ee)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

SebastianBoe committed:
[nrf noup] cmake: multi image build support with IMAGE prefix

Note: This was originally submitted upstream in #13672 and was
      rejected. NCS is carrying it as a noup for now. Another,
      more upstream-friendly approach will be needed.

We are seeing the need to boot Zephyr applications in multiple
stages, a real-world use-case of even a 3-stage bootloader has
been identified and tested.

Each bootloader needs to be individually upgrade-able and have
it's own configurations of Zephyr libraries. To achieve this each
bootloader is organized as it's own executable.

The problem is that the build system can only build one
executable. This creates a usability issue as the user must invoke a
large set of arcane commands to build, sign, and flash each
bootloader.

To resolve this we re-organize the build system such that it can build
multiple executables. To work within CMake restrictions that logical
target names must be globally unique (among other naming
restrictions), we add an IMAGE variable which is used to name the
current Zephyr application being built.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>
Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>
Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>
Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>
Signed-off-by: Ruth Fuchss <ruth.fuchss@nordicsemi.no>

(cherry picked from commit 3db22acba35aa62abb18f4823102401f8b211a57)
(Conflicts were resolved in lib/posix/CMakeLists.txt. Changes to
 ia32.cmake were dropped entirely since Intel has rejected this
 approach.

 Finally, these follow-up fixes and dependent patches were squashed
 in:

  02eeeb779665f126f59c185430d854bc7392f18e
  c6c3fed492e0f49679f5c6230dfc955c72f4ffaf
  56f6a33a71c46ec5b2e1beb6076edf9521014ad3
  1dc0ca5ccead37a53f600c65a941938f4138cf38
  6e348b8e609109981cbc140f06c324d938b52cd4)
Signed-off-by: Marti Bolivar <marti.bolivar@nordicsemi.no>

(cherry picked from commit 1e0a3dc603d9cbb9969563f1cbf308b147582461)
(Conflict resolved in cmake/app/boilerplate.cmake)
Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>
(cherry picked from commit 439a1ad80b7618ae58477fe75477670d44f654d8)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

rlubos committed:
[nrf noup]: net: socket: Add SO_RCVTIMEO and SO_BINDTODEVICE options

Socket options symbols were recently introduced upstream, now we just
need to complement them with the ones we need.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>
Signed-off-by: Marti Bolivar <marti.bolivar@nordicsemi.no>
(cherry picked from commit 77ca277934504be11815d8e4636bb2a00a284464)
(cherry picked from commit b672601e0457cd9b68cb3a4b89a9e02c778431cb)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

nvlsianpu committed:
[nrf noup] ci: Applied fix for compliance script parameters

Only pull requests will receive parameters for the comparison of
commits.
Otherwise will compare with the origin.

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>
(cherry picked from commit 175d7f36db354003e3ec76aba5ba61d2cc9db6d1)
(cherry picked from commit 39c0ac9ae27fd7e49d2319a358384851963ae7c8)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

carlescufi committed:
[nrf noup] ci: Adapt Jenkinsfile to zephyr modules

Now that we use Zephyr modules, build upstream only in
fw-nrfconnect-zephyr CI.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>
(cherry picked from commit 578dac80f7c3bafb3d0e98c0c31b072139251395)
(cherry picked from commit 7657a9d1bf52f1d1573bff1c491e96696c97c439)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

gbakke committed:
[nrf noup]: include: net: Align with bsdlib defines.

Adding socket family type AF_LOCAL.
Adding socket type SOCK_MGMT used in conjuction with with AF_LTE.
Adding socket protocol NPROTO_PDN to be used in conjuction with AF_LTE.
Adding socket protocol NPROTO_DFU to be used in conjuction with
AF_LOCAL.

Adding SOL_PDN and SOL_DFU socket option levels, and associated socket
option values.

Signed-off-by: Glenn Ruben Bakke <glenn.ruben.bakke@nordicsemi.no>
Signed-off-by: Marti Bolivar <marti.bolivar@nordicsemi.no>
(cherry picked from commit 6a62db59a215614f43ed2d8b13b6bef427acc2e2)
(cherry picked from commit 6702fed82571fed6f7722a8c3f28a720616121db)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

rlubos committed:
[nrf noup] jenkinsfile: Update NCS toolchain version

CI does not pass with the old NCS toolchain.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>
(cherry picked from commit 53d2d0448e6bd850b3d86a0b54f687183afb78ec)
(cherry picked from commit 15c08be0902d68ce81052389de8e5474ba1939fb)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

oyvindronningstad committed:
[nrf temphack] cortex_m: linker: Add ext_abi section

This is used by the immutable bootloader for sharing APIs between
images.

This hack will be reverted when zephyr_linker_sources() is available.
(Currently PR # 15112)

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>
(cherry picked from commit 3757f8cfe6ffe2c2b537815bb9ca1ace85882df8)
(cherry picked from commit cfe9b053fe1c0599ad4c3164fc899a68ff2a02e8)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

hakonfam committed:
[nrf noup] cmake: automatically change board from xxx_ns to xxxx

This is done so that sub-images of a non-secure image
is built as secure.

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>
Signed-off-by: Jan Tore Guggedal <jantore.guggedal@nordicsemi.no>

(cherry picked from commit 82bf9c7fa7d498590aa26160b83fbee13be4d5ab)
Signed-off-by: Marti Bolivar <marti.bolivar@nordicsemi.no>

(cherry-picked from commit 7e567d75777b4cb2498a8d7a0938a2e46a884737)
Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>
(cherry picked from commit 0a85bddb1bb3e01063316429521e4d5c55b2b0df)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

carlescufi committed:
[nrf noup] jenkinsfile: Multiple fixes

- In order to ensure that we use a compatible revision of ci-tools, fix
the SHA of the repo to a version we know is compatible.

- Use the Zephyr SDK instead of the GNU Arm Embedded toolchain.

- Add retries on sanitycheck for unstable builds

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>
(cherry picked from commit f3c37a804dac59a6e06002691fa6006333001d69)
(cherry picked from commit 47aa0276f698718e656fa1796d15f83f01152910)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

hakonfam committed:
[nrf toup] cmake: only unset CMAKE_C_COMPILER for first image

The C compiler does not change between the different images.
Add a check to ensure that the C compiler variable is only
unset for the first boilerplate execution.

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>
(cherry picked from commit 33c06133c197dbf423d36ffc6f582c736dda92d7)
(cherry picked from commit 6c85501e178d835bd2a18c222732ba639953a5c2)
(cherry picked from commit 05270eab2d561870347d36844485869229d20de8)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

sigvartmh committed:
[nrf noup] partition manager: add mandatory zephyr pieces

Partition Manager is a component which uses yaml files to
resolve flash placement with a wholistic view of the
device

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>
Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>
Signed-off-by: Sigvart Hovland <sigvart.hovland@nordicsemi.no>
Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>
Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

(cherry picked from commit 72d11e22210adba16d45862bcc84cbcc3d09275b)
Signed-off-by: Marti Bolivar <marti.bolivar@nordicsemi.no>

(cherry picked from commit 68ec90c85806870700170a0b5cafbc468a36d7a8)
(Conflict resolved in include/arch/arm/cortex_m/scripts/linker.ld)
Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>
(cherry picked from commit 03c296db23539158553502346434abb9c667317c)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

ChristopherBittner committed:
[nrf noup] ci: limit build agent selection

Only build agent should be used for building jobs.

Signed-off-by: Chris Bittner <chris.bittner@nordicsemi.no>
(cherry picked from commit 4498555744c0a4cdbb2c11f72b6f69b5433658b1)
(cherry picked from commit eff73e79529183062774492bc75d0910d756a629)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

thst-nordic committed:
[nrf noup] ci: add linux label to jenkins agent

Required now that we have windows agents.
also fix stray tabs->spaces

Signed-off-by: Thomas Stilwell <Thomas.Stilwell@nordicsemi.no>
(cherry picked from commit 98e9dc773f999404669bcfffadf7406e164bb434)
(cherry picked from commit a7c0185f812ff74311d54f884c98b73cae635561)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

thst-nordic committed:
[nrf noup] ci: move ci-tools to latest on master

Fixes some missing KConfig argument errors in check_compliance.py

Signed-off-by: Thomas Stilwell <Thomas.Stilwell@nordicsemi.no>
(cherry picked from commit 55cada4fb79ff74e08f3369f78ebd78faafb24aa)
(cherry picked from commit 11148f32101725c225c2317f305acbc708ad7984)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

oyvindronningstad committed:
[nrf toup] arch: arm: tz: Add wrapper target for secure entry veneers

This is needed for make to resolve dependencies correctly.
Also refactor a bit.

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

(cherry picked from commit f19f72d8e3f62ccd5dc7fbcfa45bcbf7190aefd8)
Signed-off-by: Marti Bolivar <marti.bolivar@nordicsemi.no>

(cherry picked from commit c5d876135d6ef48347887ddf23b7ec30077092be)
Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>
(cherry picked from commit 227db55bf7750b141e57a166fbd4193a9ad69026)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

thst-nordic committed:
[nrf noup] ci: allow jenkins build plans on tags

Use {env.TAG_NAME} in jenkinsfile to find correct COMMIT_RANGE

Signed-off-by: Thomas Stilwell <Thomas.Stilwell@nordicsemi.no>
(cherry picked from commit 330fd0fbea345a6ba59eb893902c92f41d579817)
(cherry picked from commit 4aa8728de4baf28fca5ec1246f0ff5089270c48b)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

thst-nordic committed:
[nrf noup] ci: Skip Kconfig compliance check temporarily

This is a temporary fix. We will find a permanent solution after
v1.0.0 release.

Signed-off-by: Thomas Stilwell <Thomas.Stilwell@nordicsemi.no>
(cherry picked from commit 8ff27f817746b8d61637fe7ac8a5faebbe348a8f)
(cherry picked from commit 98144c96dba9b67e7da0b3deed5cc4bc2aa1da28)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

oyvindronningstad committed:
[nrf noup] linker.ld: Change ifdef from SECURE_BOOT to FW_METADATA

FW_METADATA is its own subsys now.
This improves user experience with B0 + MCUboot.

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>
(cherry picked from commit 04b29a741e42a9844afeb7e144d1f6e980979234)
(cherry picked from commit c1f74954e368ab88aa8a33361c431e52a6ba689d)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

mbolivar-nordic committed:
[nrf temphack] cmake: carry forward dts.cmake patches from a mergeup

Mergeup commit 1378a1c7ac89d13fa08bd45b1272a70060d90663 made changes
to cmake/dts.cmake which I'm carrying forward in a separate patch for
now.

Need more clarity from Robert or Sebastian on what "ought" to happen to
this patch:

     1. squash into multi-image?
     2. submit upstream?

Signed-off-by: Marti Bolivar <marti.bolivar@nordicsemi.no>
Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>
Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>
(cherry picked from commit b5e975ad8a0e831e1a8dc582db73c5a7f306d39c)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

thst-nordic committed:
[nrf noup] ci: use Standard Library in Jenkinsfile

Managing Docker version, Agent Labels, Github Status,

manifest updates, downstream jobs from CI_LIB.

Signed-off-by: Thomas Stilwell <Thomas.Stilwell@nordicsemi.no>
(cherry picked from commit 9420e7247ba2827f58940a921b3837bf3a2f8cd8)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

thst-nordic committed:
[nrf noup] Jenkinsfile: Build PR only on nRF platform

PR should be only build on nRF platform to
save a time. Master and other branch are tested
in default configuration.

Signed-off-by: Kamil Gawor <Kamil.Gawor@nordicsemi.no>
Signed-off-by: Thomas Stilwell <Thomas.Stilwell@nordicsemi.no>
(cherry picked from commit 71b802740d6d9a57697e5f0c6e76bb39f8a0caa4)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

thst-nordic committed:
[nrf noup] ci: adding parallel sanitycheck steps

only 1 of the parallel stages is run

Signed-off-by: Thomas Stilwell <Thomas.Stilwell@nordicsemi.no>
(cherry picked from commit ec52cc75e99ca715f8d6d5c0aed906a79b1e2879)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

SebastianBoe committed:
[nrf fromlist] samples: Added a minimal MCUBoot integration test

From upstream #13672

Added a minimal integration test for MCUBoot. It has just enough
coverage to catch build issues. Which is better than before (no
coverage in the Zephyr CI).

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>
(cherry picked from commit 1937b41459ca722848abe88cf84067945e8cf0c2)
(cherry picked from commit 026bd218e410fd679baafb2d4b4e95657539a917)
(cherry picked from commit 4b3a15bd6a793dfe23117684e6fa65a82c8fe12e)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

mike-scott committed:
[nrf noup] net: lwm2m: add timestamp fields to several IPSO obj

Let's add a timestamp field to the following objects for better
event tracking:
- IPSO Accelerometer
- IPSO On/Off switch
- IPSO Push Button
- IPSO Temperature

Signed-off-by: Michael Scott <mike@foundries.io>
(cherry picked from commit bd9a3605f0c8a276f89fbf15940ed053b6e37a14)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

carlescufi committed:
[nrf temphack] Bluetooth: controller: Default to legacy

Default to the legacy controller since our downstream apps won't build
with the nRF or Split controllers yet.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>
(cherry picked from commit 9de1096e041e302f2da57b319a207c8bc60e4d4b)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

hakonfam committed:
[nrf toup] drivers/flash: leverage nsc flash read

When current image is non-secure and address being read is configured
as secure, use the Non Secure Callable function spm_request_read to
try and read the secure flash area.

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>
(cherry picked from commit 8fd3c7817175639213d03e74375242844f479fca)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

hakonfam committed:
[nrf noup] kconfig: add option for building s1 variant

Also enforce this option on mcuboot.

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>
(cherry picked from commit 6be5ea58ebf378be6021214296b50e6b386cae05)
(cherry picked from commit 66585a024cb316a3384f4bb591a44360bc423717)
Signed-off-by: Marti Bolivar <marti.bolivar@nordicsemi.no>

(cherry-picked from commit 611c2fcb801e91077b79c16d937bf0c79e064b98)
(squashed in the following one-line fixes:
	02fb10778e89e3365659e58ae218c9a9f63fdc5d
	e9236d5d0f625679b164e5d039f2464aacd1f5ed
	217212054fbbb464d573594c3eaf4d775cea2d55)
Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

(cherry picked from commit 47df92b83dd402531725319d6b06f5f2063b7ddc)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

hakonfam committed:
[nrf noup] linker: add symbol for image 1 primary slot

This symbol is used to ensure that s0 mcuboot points to s1 as
image 1 primary slot, and s1 mcuboot points to s0 as image 1
primary slot.

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>
(cherry picked from commit 40b430ba977cfea2cf733248d79d99ef0b57c19e)

(cherry picked from commit ff7db1214cdfaaebd023759e6fa80189b0a73539)
Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>
(cherry picked from commit 50303b87794d066b4e47c5c4da4550c5f58d8493)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

alextsam committed:
[nrf noup] cmake: automatically change board from xx_ns to xx (Icarus)

Patch boilerplate for supporting also the Actinius Icarus board

Signed-off-by: Alex Tsamakos <alex@actinius.com>
(cherry picked from commit 6c98a692496745e52631bc03889eca06db18194f)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

hakonfam committed:
[nrf toup] drivers/flash: limit secure read inclusion

Only include secure read call if SPM is enabled and partition manager is
in use.

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>
(cherry picked from commit b3bc959667dc171f8c4087c900c19a7c0a458e0c)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

thst-nordic committed:
[nrf noup] ci: build NRF sanitycheck on all builds'

run NRF and upstream sanitycheck in parallel

Signed-off-by: Thomas Stilwell <Thomas.Stilwell@nordicsemi.no>
(cherry picked from commit 4493a423a645cb54de2ef2720ebb7c403774ae11)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

Commit:
[nrf noup]: include: net: Add option for PDN sockets.

Adding SO_PDN_CONTEXT_ID option value for PDN sockets.

Signed-off-by: Christopher Métrailler <christopher.metrailler@nordicsemi.no>
(cherry picked from commit 71dcbe440a732b052424c8dfe9129c28fbb054db)
(cherry picked from commit d83de9040a426051e65650aeb4b70cfd97744027)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

thst-nordic committed:
[nrf noup] ci: force gnuarmemb sanitycheck and run in parallel

jenkins now load balances the parallel tasks better also.

Signed-off-by: Thomas Stilwell <Thomas.Stilwell@nordicsemi.no>
(cherry picked from commit 05f1cb3f9201f019d97b0890bf137939c074bc9e)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

SebastianBoe committed:
[nrf noup] cmake: Port lib/posix to multi-image

Port lib/posix to be compatible with multi-image.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>
(cherry picked from commit 76d88f1ccc7d5fb89da811db793a002136f63f25)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

lemrey committed:
[nrf noup] net: socket: add SO_DFU_ERROR socket option

Add SO_DFU_ERROR socket option for DFU socket.
The option can be used when a operation on a DFU socket
returns -ENOEXEC, indicating that the modem has rejected
the operation, to retrieve the reason for the error.

Signed-off-by: Emanuele Di Santo <emdi@nordicsemi.no>
(cherry picked from commit 6f06005cd5e1733fdcb6c3f3d0b90d16df640450)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

lemrey committed:
[nrf noup] net: socket: add SO_PDN_STATE socket option

Add SO_PDN_STATE option for PDN sockets
to retrieve the state of the PDN connection.

Signed-off-by: Emanuele Di Santo <emdi@nordicsemi.no>
(cherry picked from commit 57bca6dd70f67b50a0659ed379c10f4a5b758696)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

LeHack committed:
[nrf noup] ci: Run all sanitychecks on every PR using zephyr compiler.

From now all Zephyr PRs will execute all sanitycheck tests using.
Additionally we will use the zephyr compiler for non-nrf sanitychecks.

Signed-off-by: Łukasz Hejnak <lukasz.hejnak@nordicsemi.no>
(cherry picked from commit b645105871b349a1cc8a7c168ee9d00a25afd74e)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

lemrey committed:
[nrf noup] net: socket: fix SO_DFU_RESOURCES definition

Change from SO_DFU_RESOURCE to SO_DFU_RESOURCES.

Signed-off-by: Emanuele Di Santo <emdi@nordicsemi.no>
(cherry picked from commit d6e67554cfeb28df5525e53c073f61a00836f283)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

oyvindronningstad committed:
[nrf noup] cortex_m: linker.ld: Rename FW_METADATA to FW_INFO

Rename configs used in linker script

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

ioannisg committed:
[nrf toup] tests: drivers: counter: Add clock stabilization nRF

Xtal LF clock source starts hundreds of milliseconds. When it is
not start, test may fail due to wrong timing. Added pending
on LF clock being start in test setup.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>
Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>
(cherry picked from commit 94b7d5affeea5df700c2aa1e3329292fa3b8e8eb)
(cherry-picked from commit 7cf799102d9a35781cc09b2f1d515ec2386ad1c4)
Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>
(cherry picked from commit ade600a816779bb71f2e94e0f979b1552d6a9352)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

tejlmand committed:
[nrf fromlist] cmake: zephyr_library_amend feature

This commit introduces the cmake extension zephyr_library_amend.

This function allows for adding files in an out-of-tree Zephyr module
to a zephyr library created in zephyr repo CMake files.

As example:
drivers/entropy/CMakeLists.txt creates an zephyr library as:
zephyr_library()
only available to zephyr itself.

The amend function allows to amend to such a lib, by creating a
CMakeLists.txt file following identical folder structure in a Zephyr
Module:
<zephyr_module_oot>/drivers/entropy/CMakeLists.txt
zephyr_library_amend()
zephyr_library_sources() # Sources are amended to the original library

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>
(cherry picked from commit 5e35b0c90196760b713e00004804d9de8a62282e)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

thst-nordic committed:
[nrf noup] ci: skip compliance check on commits after [nrf mergeup]

run normal if not mergeup PR

Signed-off-by: Thomas Stilwell <Thomas.Stilwell@nordicsemi.no>
(cherry picked from commit 8e147e602af36da7d74b1248ed472df08229e4e6)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

thst-nordic committed:
[nrf noup] ci: report back to github on UNSTABLE ci result

send FAILURE because github have UNSTABLE

Signed-off-by: Thomas Stilwell <Thomas.Stilwell@nordicsemi.no>
(cherry picked from commit 1c29447c9b4079e47b13d89031f1c098057e0540)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

tejlmand committed:
[nrf noup] cmake: multiimage: zephyr_library_amend feature

This commits adds ${IMAGE} in order to support multiimage in the amend
feature.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>
(cherry picked from commit 3a0c666ed482a4a03b3175cc81bec230f2ca08c6)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

thst-nordic committed:
[nrf toup] sanitycheck: append --board-root args

this was removed in sanitycheck refactor

Signed-off-by: Thomas Stilwell <Thomas.Stilwell@nordicsemi.no>
(cherry picked from commit ede6735dd84e050965a8516d33848c2f1d930721)
Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

sofu1 committed:
bluetooth: controller: Include nrf_clock_control.h when MPSL is used

Enable use of defines in nrf_clock_control.h when MPSL clock driver
is used in Zephyr bluetooth LL.

Signed-off-by: Solveig Fure <solveig.fure@nordicsemi.no>

rugeGerritsen commented on subsys/bluetooth/controller/ll_sw/ll.c:
```diff
@@ -11,7 +11,7 @@
 #include <device.h>
 #include <drivers/clock_control.h>
-#ifdef CONFIG_CLOCK_CONTROL_NRF
+#if defined(CONFIG_CLOCK_CONTROL_NRF) || defined(CONFIG_CLOCK_CONTROL_MPSL)
```
I don't think we can get change to upstream zephyr. We should also try to prevent changes to zephyr, as this will take more time to get through. Do we need this change, or can we reuse CONFIG_CLOCK_CONTROL_NRF?

sofu1 replied:
Would it be ok to just do it to the fw-nrfconnect-zephyr? I was not able to make it work with CONFIG_CLOCK_CONTROL_NRF. I can try again though if this is a problem.

rugeGerritsen replied:
I think we would try to avoid that if possible. Otherwise the two repos will diverge, which will increase maintenance cost

sofu1 replied:
Yes, the problem is that zephyr sets nrf_power_clock.c as source if CONFIG_CLOCK_CONTROL_NRF is defined.

sofu1 replied:
https://github.com/zephyrproject-rtos/zephyr/pull/20134

nvlsianpu committed:
[nrf fromtree] Test: Settings: Fix FCB delete test

This commit fixes the FCB delete test after PR #19541.
Now the entity callback is not called on deleted element.

Signed-off-by: Radoslaw Koppel <radoslaw.koppel@nordicsemi.no>
Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

thst-nordic committed:
[nrf noup] ci: use ci-tools in west manifest

NCSDK-3166: re-run sanitycheck correctly on QEMU failure

NCSDK-3134: add PENDING/UNSTABLE github status reports
Signed-off-by: Thomas Stilwell <Thomas.Stilwell@nordicsemi.no>

nvlsianpu committed:
[nrf fromlist] tests/settings/functional/file: fix nRF targets

Pointed nRF flash driver for the NFFS.
Before improper flash driver name caused early test failure.

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>
